helm/dex-app/templates/dex-k8s-authenticator/deployment-giantswarm.yaml

```diff
@@ -83,22 +83,32 @@ spec:
         configMap:
           name: {{ include "resource.dexk8sauth.name" . }}-giantswarm
       {{- if not .Values.ingress.tls.letsencrypt }}
+      {{- if .Values.trustedRootCA }}
```

Comment:
Just for my interest, what is the problem we are solving here?
It appears that we add the CA to the giantswarm connectors dex-k8s-authenticator instance.
However, from my understanding, gs staff does not use dex-k8s-authenticator. Since we log in via kgs, shouldn't we receive the trusted CA cert directly from athena?
In cutomers case we also do not recommend using dex-k8s-authenticator anymore and we want to get rid of it.

Reply:
We need to solve 2 problems:

1. The problem that we solve is that the dex-k8s-authenticator pod crashes without a valid trusted root CA.
The authenticator needs to communicate with Dex over https. And the authenticator fails and crashes in case Dex uses certificate signed by a CA, which is not trusted by the authenticator. Even though the authenticator no longer actively participates in the authentication process, I think it should still at least run as long as it's part of the Dex app installation.

2. The CA is supposed to be included in the secret with certificates for the ingress. I don't think the CA is needed for the ingress (although I am not entirely sure about that). In some cases the certificates in that secret are provided/overwritten by the cert manager but it is not able to provide the CA. We need to ensure that the certificates provided by the cert manager are not overwritten by re-deployments of the Dex app. The current state of the PR is based on an assumption that the cert manager overwrites the certificates after every re-deployment of the Dex app, but it needs to be verified, and if it's not the case, then it needs to be addressed.

Reply:
The change is not specific to Giantswarm staff authentication. It needs to be applied to customer authenticator deployment, too.
It should be fixed now.

Reply:
Thank you for the explanation!
With adding it to the customer connector it makes sense for me now.